## Reddit Toxicity Detection

This dataset contains posts from various subreddits, with each entry providing information on the subreddit name, the text content of the post (body), its controversiality (indicating if it sparked significant discussion), and the score (upvotes minus downvotes) received by the post. It can be used to analyze user engagement and sentiment across different subreddits.

### The Dataset has following attributes
1. **subreddit**: The name of the subreddit where the post was made.
2. **body:** The content or text of the post or comment.
3. **controversiality:** Indicates whether the post is controversial (1 for controversial, 0 for not).
4. **score:** The score (upvotes minus downvotes) received by the post.


In [2]:
import pandas as pd
import numpy as np

In [6]:
## read the dataset
df = pd.read_csv("data/kaggle_RC_2019-05.csv")
df.head()

,subreddit,body,controversiality,score
0,gameofthrones,Your submission has been automatically removed...,0,1
1,aww,"Dont squeeze her with you massive hand, you me...",0,19
2,gaming,It's pretty well known and it was a paid produ...,0,3
3,news,You know we have laws against that currently c...,0,10
4,politics,"Yes, there is a difference between gentle supp...",0,1


In [7]:
## shape of the data
df.shape

(1000000, 4)

In [8]:
## null values
df.isnull().sum()

subreddit           0
body                0
controversiality    0
score               0
dtype: int64

There are no null values

In [14]:
## Different subreddit categorirs
print(df['subreddit'].unique())
print("\nTotal number of distinct subreddit categories: ",df['subreddit'].nunique())

['gameofthrones' 'aww' 'gaming' 'news' 'politics' 'dankmemes'
 'relationship_advice' 'nba' 'worldnews' 'AskReddit' 'AmItheAsshole'
 'SquaredCircle' 'The_Donald' 'leagueoflegends' 'hockey' 'videos'
 'teenagers' 'gonewild' 'movies' 'funny' 'pics' 'marvelstudios' 'memes'
 'soccer' 'freefolk' 'MortalKombat' 'todayilearned' 'apexlegends' 'asoiaf'
 'Market76' 'Animemes' 'FortNiteBR' 'nfl' 'trashy' 'unpopularopinion'
 'ChapoTrapHouse' 'RoastMe' 'Showerthoughts' 'wallstreetbets' 'Pikabu']

Total number of distinct subreddit categories:  40


In [15]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000000 entries, 0 to 999999
Data columns (total 4 columns):
 #   Column            Non-Null Count    Dtype 
---  ------            --------------    ----- 
 0   subreddit         1000000 non-null  object
 1   body              1000000 non-null  object
 2   controversiality  1000000 non-null  int64 
 3   score             1000000 non-null  int64 
dtypes: int64(2), object(2)
memory usage: 30.5+ MB


In [16]:
df.describe()

,controversiality,score
count,1000000.000000,1000000.000000
mean,0.029583,11.510103
std,0.169434,149.671560
min,0.000000,-889.000000
25%,0.000000,1.000000
50%,0.000000,2.000000
75%,0.000000,4.000000
max,1.000000,35619.000000


There are a total of 1 million rows.
1. controversiality is a binary attribute that has either 0 or 1
2. score ranges from -889 to a maximum of 35619 which a total of upvotes and down votes for each commenter